File: app/src/main/java/com/example/sampleautoscreenoffwhennointeraction/MyBaseActivity.java
package com.example.sampleautoscreenoffwhennointeraction;

import android.app.Activity;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.os.PowerManager;
import android.util.Log;
import android.view.WindowManager;


public class MyBaseActivity extends Activity {

    public static final long DISCONNECT_TIMEOUT = 5000; // 300000 -> 5 min = 5 * 60 * 1000 ms

    /*private static Handler disconnectHandler = new Handler(new Handler.Callback() {
        @Override
        public boolean handleMessage(Message msg) {
            // todo
            return true;
        }
    });*/

    private void turnOnScreen(){
        // turn on screen
        WindowManager.LayoutParams params = getWindow().getAttributes();
        params.screenBrightness = -1;
        getWindow().setAttributes(params);
    }

    private void turnOffScreen(){
        // turn off screen
        WindowManager.LayoutParams params = getWindow().getAttributes();
        params.screenBrightness = 0;
        getWindow().setAttributes(params);
    }

    private final Handler disconnectHandler = new Handler(Looper.getMainLooper());

    private Runnable disconnectCallback = new Runnable() {
        @Override
        public void run() {
            // Perform any required operation on disconnect
            turnOffScreen();
        }
    };

    public void resetDisconnectTimer(){
        disconnectHandler.removeCallbacks(disconnectCallback);
        disconnectHandler.postDelayed(disconnectCallback, DISCONNECT_TIMEOUT);
    }

    public void stopDisconnectTimer(){
        disconnectHandler.removeCallbacks(disconnectCallback);
    }

    @Override
    public void onUserInteraction(){
        resetDisconnectTimer();
        turnOnScreen();
    }

    @Override
    public void onResume() {
        super.onResume();
        resetDisconnectTimer();
    }

    @Override
    public void onStop() {
        super.onStop();
        stopDisconnectTimer();
    }
}

File: app/src/main/java/com/example/sampleautoscreenoffwhennointeraction/MyBaseActivity.kt
package com.example.sampleautoscreenoffwhennointeraction

import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity

open class MyBaseActivity : AppCompatActivity() {

    companion object {
        const val DISCONNECT_TIMEOUT: Long = 5000 // 300000 -> 5 min = 5 * 60 * 1000 ms
    }

    private fun turnOnScreen() {
        // turn on screen
        val params = window.attributes
        params.screenBrightness = -1f
        window.attributes = params
    }

    private fun turnOffScreen() {
        // turn off screen
        val params = window.attributes
        params.screenBrightness = 0f
        window.attributes = params
    }

    private val disconnectHandler = Handler(Looper.getMainLooper())
    private val disconnectCallback = Runnable {
        // Perform any required operation on disconnect
        turnOffScreen()
    }

    fun resetDisconnectTimer() {
        disconnectHandler.removeCallbacks(disconnectCallback)
        disconnectHandler.postDelayed(disconnectCallback, DISCONNECT_TIMEOUT)
    }

    fun stopDisconnectTimer() {
        disconnectHandler.removeCallbacks(disconnectCallback)
    }

    override fun onUserInteraction() {
        resetDisconnectTimer()
        turnOnScreen()
    }

    public override fun onResume() {
        super.onResume()
        resetDisconnectTimer()
    }

    public override fun onStop() {
        super.onStop()
        stopDisconnectTimer()
    }

}